tests/bulk-tx/src/bulk_tx.ts

```diff
@@ -0,0 +1,144 @@
+import { ApiPromise, WsProvider, Keyring } from '@polkadot/api';
+import { SignerResult } from '@polkadot/api/types';
+import { KeyringPair } from '@polkadot/keyring/types';
+import type { EventRecord, ExtrinsicStatus, H256, SignedBlock } from '@polkadot/types/interfaces';
+import type { ISubmittableResult, SignatureOptions } from '@polkadot/types/types';
+
+const ALICE = '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY';
+const keyring = new Keyring({ type: 'sr25519' });
+
+//batch size and payload size are set here
+const batch:number = 3;
+const size:number = 100;
+
+async function createApi(): Promise<ApiPromise> {
+  // Initialise the provider to connect to the local node
+  // const provider = new WsProvider('wss://polygon-da-explorer.matic.today/ws');
+  const provider = new WsProvider('ws://127.0.0.1:9944');
+
+  // Create the API and wait until ready
+  return ApiPromise.create({
+    provider,
+    types: {
+      DataLookup: {
+        size: 'u32',
+        index: 'Vec<(u32,u32)>'
+      },
+      KateExtrinsicRoot: {
+        hash: 'Hash',
+        commitment: 'Vec<u8>',
+        rows: 'u16',
+        cols: 'u16'
+      },
+      KateHeader: {
+        parentHash: 'Hash',
+        number: 'Compact<BlockNumber>',
+        stateRoot: 'Hash',
+        extrinsicsRoot: 'KateExtrinsicRoot',
+        digest: 'Digest',
+        appDataLookup: 'DataLookup'
+      },
+      Header: 'KateHeader',
+      AppId: 'u32',
+      // CheckAppId: {
+      //     extra: {
+      //         appId: 'u32', 
+      //     },
+      //     types: {}
+      // }
+    },
+    signedExtensions: {
+      CheckAppId: {
+        extrinsic: {
+          appId: 'u32'
+        },
+        payload: {}
+      },
+    },
+  });
+}
+
+//generating random data as per payload specified
+function randomDigit() {
+  return Math.floor(Math.random() * Math.floor(2));
```

Comment:
Why are we using `Math.floor(2)`?

Reply:
To get the numbers 0 & 1. So effectively its like %2 to every number we multiplied with `math.random()`

Reply:
I understand that. But `Math.floor(2)` equals `2` always! Is there a special reason to use the floor function?

Reply:
So actually the result is different when using simple 2 and math.floor(2) here. Thats what I found

Reply:
oh nvm mind! Its the same! My mistake on it 😅